src/index.ts

```diff
@@ -160,14 +159,14 @@ export function objectsEqualShallow(
  * @template T,R
  */
 export function memo<T, P extends keyof T>(
```

Comment:
see https://github.com/tc39/proposal-function-memo

Reply:
Yeah, i knew about that proposal, but am not yet sure how it relates to this function. Is the difference that this is Object memo and that would be Function memo?

/cc @js-choi